### ISSUE TITLE
imageBuild is broken

### ISSUE THREAD

**Issue by rodrigoaguilera** [beluga-php/docker-php#60]
I have been trying to update to the latest 1.43 latest for some weeks but I couldn't due to some cryptic error on the socket when I tried to do an imageBuild.
```
[error]  An error occur when writing request to client (BROKEN EPIPE)
ERROR: fwrite(): Send of 8192 bytes failed with errno=32 Broken pipe
in /home/basin/vendor/php-http/socket-client/src/RequestWriter.php:97
```
After quite a lot of investigation updating docker-php commit by commit I found 
https://github.com/beluga-php/docker-php/commit/c44e04675829abf5863b08182296fcfc0df1c73b

I don't know if that was an accidental removal since the commit message explains something else.
That commit makes many classes under the `\Docker\Endpoint` dead code except for ImagePush.
In the case of ImageBuild the wrapper endpoint is needed to send a proper tar stream to the socket.
Also I don't know if the other calls like `containerLogs` are broken too.

I believe the action plan would be to first restore those lines in Docker.php and then add some kind of end-to-end tests to ensure imageBuild won't be broken in the future.

**Comment by flavioheleno** [comment]
First of all, I'm so sorry for the extra work you had to go through, it was def **not** my intention!

About the commit, I actually did it with some reasoning, but now I see where it failed.

Let me be a bit clear around my intentions:

I saw this in `src/Docker.php`:

```php
public function containerAttach(
    string $id, 
    array $queryParameters = [], 
    string $fetch = self::FETCH_OBJECT, 
    array $accept = []
) {
    return $this->executeEndpoint(new ContainerAttach($id, $queryParameters, $accept), $fetch);
}
```

And this in `src/Client.php` (from `docker-php-api`):

```php
public function containerAttach(
    string $id, 
    array $queryParameters = [], 
    string $fetch = self::FETCH_OBJECT, 
    array $accept = []
) {
    return $this->executeEndpoint(
        new \Docker\API\Endpoint\ContainerAttach($id, $queryParameters, $accept), 
        $fetch
    );
}
```

And obviously, due to lack of attention, I took `Docker\Endpoint\ContainerAttach` as equivalent to `\Docker\API\Endpoint\ContainerAttach` - I know, stupid right?

**TLDR**: I had good intentions but broke the code. Sorry! :-(

**Comment by flavioheleno** [comment]
> I believe the action plan would be to first restore those lines in Docker.php and then add some kind of end-to-end tests to ensure imageBuild won't be broken in the future.

I agree 100% with you on that! Gotta do some joint effort to achieve that though.

**Comment by rodrigoaguilera** [comment]
No problem. I was not my intention to be read as angry or anything like that. I'm actually quite happy to have figured out the issue.

I won't open the issue about the end-to-end tests until I investigate a bit and I can come with a minimal proposal. 

Thanks for reviewing the PR :)

**Comment by flavioheleno** [comment]
> I was not my intention to be read as angry or anything like that.

I didn't read as it either :-) I just feel really bad when I give unnecessary extra work to people.
